contracts/validators/CredentialAtomicQueryValidator.sol

```diff
@@ -109,17 +130,61 @@ abstract contract CredentialAtomicQueryValidator is OwnableUpgradeable, ICircuit
 
                 if (
                     block.timestamp - claimNonRevLatestStateInfo.replacedAtTimestamp >
-                    revocationStateExpirationTime
+                    revocationStateExpirationTimeout
                 ) {
                     revert("Non-Revocation state of Issuer expired");
                 }
             }
         }
     }
 
-    function _checkProofGeneratedExpiration(uint256 _proofGenerationTimestamp) internal view {
-        if (block.timestamp - _proofGenerationTimestamp > proofGenerationExpirationTime) {
+    function _checkProofExpiration(uint256 _proofGenerationTimestamp) internal view {
+        if (_proofGenerationTimestamp > block.timestamp) {
+            revert("Proof generated in the future is not valid");
+        }
+        if (block.timestamp - _proofGenerationTimestamp > proofExpirationTimeout) {
             revert("Generated proof is outdated");
         }
     }
+
+    function _checkAllowedIssuers(uint256 issuerId, uint256[] memory allowedIssuers) internal pure {
+        // empty array is 'allow all' equivalent - ['*']
+        if (allowedIssuers.length == 0) {
+            return;
+        }
+
+        for (uint i = 0; i < allowedIssuers.length; i++) {
+            if (issuerId == allowedIssuers[i]) {
+                return;
+            }
+        }
+
+        revert("Issuer is not on the Allowed Issuers list");
+    }
+
+    function _checkMerklized(uint256 merklized, uint256 queryClaimPathKey) internal pure {
+        uint256 shouldBeMerklized = 0;
+        if (queryClaimPathKey != 0) {
+            shouldBeMerklized = 1;
+        }
```

Comment:
Would ternary operator be better to write the same?

Reply:
+1